src/traces/parcoords/calc.js

```diff
@@ -0,0 +1,49 @@
+/**
+* Copyright 2012-2017, Plotly, Inc.
+* All rights reserved.
+*
+* This source code is licensed under the MIT license found in the
+* LICENSE file in the root directory of this source tree.
+*/
+
+'use strict';
+
+var hasColorscale = require('../../components/colorscale/has_colorscale');
+var calcColorscale = require('../../components/colorscale/calc');
+var Lib = require('../../lib');
+
+
+module.exports = function calc(gd, trace) {
+    var inputDimensions = trace.dimensions,
+        dimensions = [];
+    for(var i = 0; i < inputDimensions.length; i++) {
```

Comment:
> Any way to preserve your authorship of this commit?

I don't care about getting an extra commit merge. Don't waste time on this.

> Should my next feature branch e.g. Sankey simply be in the plotly repo? 

Yes please. Moreover, please don't `git rebase master` on branches associated with PRs. It makes it very difficult to keep track of commits and discussion items. Please use `git merge master` instead. 

> Minor note, I needed to return a bit more from calc.js b/c 3 unit tests failed

By unit tests, you mean those simply looking for fields on `gd`? That's a bad excuse, you can easily change those.